File: Python/order/create_order_checkout.py
from pagarmecoreapi.pagarmecoreapi_client import PagarmecoreapiClient
from pagarmecoreapi.models import *
from pagarmecoreapi.controllers import *
from pagarmecoreapi.exceptions.error_exception import *

# Configuration parameters and credentials
basic_auth_user_name = 'basic_auth_user_name' # The username to use with basic authentication
basic_auth_password = 'basic_auth_password' # The password to use with basic authentication

PagarmecoreapiClient(basic_auth_user_name, basic_auth_password)

orders_controller = orders_controller.OrdersController()

customer = create_customer_request.CreateCustomerRequest()
customer.name = "Tony Stark"
customer.email = "tonystark@avengers.com"

checkout = create_checkout_payment_request.CreateCheckoutPaymentRequest()
checkout.customer_editable = False
checkout.skip_checkout_success_page = True
checkout.accepted_payment_methods = ["credit_card", "boleto", "bank_transfer", "debit_card"]
checkout.accepted_multi_payment_methods = [["credit_card", "credit_card"], ["credit_card", "boleto"]]
checkout.success_url = "https://www.pagar.me"

# Bank transfer payment Setup
checkout.bank_transfer = create_checkout_bank_transfer_request.CreateCheckoutBankTransferRequest()
checkout.bank_transfer.bank = ["237", "001", "341"]

# Boleto Payment Setup
checkout.boleto = create_checkout_boleto_payment_request.CreateCheckoutBoletoPaymentRequest()
checkout.boleto.bank = "033"
checkout.boleto.instructions ="Pagar ate o vencimento"
checkout.boleto.due_at = "2021-07-25T00:00:00Z"

# Credit Card Payment Setup
checkout.credit_card = create_checkout_credit_card_payment_request.CreateCheckoutCreditCardPaymentRequest()
checkout.credit_card.capture = True
checkout.credit_card.statement_descriptor = "Descriptor example"
checkout.credit_card.installments = [ # Credit card installments Setup
    create_checkout_card_installment_option_request.CreateCheckoutCardInstallmentOptionRequest(),
    create_checkout_card_installment_option_request.CreateCheckoutCardInstallmentOptionRequest()]
# installment 1
checkout.credit_card.installments[0].number = 1
checkout.credit_card.installments[0].total = 2000
# installment 2 with extra tax of 500
checkout.credit_card.installments[1].number = 1
checkout.credit_card.installments[1].total = 2500

# Debit Card Payment Setup
checkout.debit_card = create_checkout_debit_card_payment_request.CreateCheckoutDebitCardPaymentRequest()
# Debit card Authentication Setup
checkout.debit_card.authentication = create_payment_authentication_request.CreatePaymentAuthenticationRequest()
checkout.debit_card.authentication.mtype = 'none'
checkout.debit_card.authentication.threed_secure = create_three_d_secure_request.CreateThreeDSecureRequest()
checkout.debit_card.authentication.threed_secure.mpi = "acquirer"
checkout.debit_card.authentication.threed_secure.success_url = "https://www.pagar.me"

request = create_order_request.CreateOrderRequest()
request.code = "test-SDK-python"
request.items = [create_order_item_request.CreateOrderItemRequest()]
request.items[0].description = "Tesseract Bracelet"
request.items[0].quantity = 1
request.items[0].amount = 2990

request.payments = [create_payment_request.CreatePaymentRequest()]
request.payments[0].amount = 2000
request.payments[0].payment_method = "checkout"
request.payments[0].checkout = checkout
request.customer = customer

try:
    result = get_order_response.GetOrderResponse()
    result.checkouts = [get_checkout_payment_response.GetCheckoutPaymentResponse()]
    result = orders_controller.create_order(request)
    print("Order id: ", result.id)
    print("Checkout id: ", result.checkouts[0].id)
    print("Order result status: ", result.status)
    print("Checkout payment url: ", result.checkouts[0].payment_url)
    assert result.status == "pending"
    assert result.checkouts[0].boleto is not None
    assert result.checkouts[0].credit_card is not None
    assert result.checkouts[0].debit_card is not None
    assert result.checkouts[0].bank_transfer is not None
    assert result.checkouts[0].payment_url is not None
except ErrorException as ex:
    print("\n")
    print("Error Message: ", ex.message)
    print("Status code Response: ", ex.response_code)
    if(ex.errors is not None):
        print("Errors: ", ex.errors)
except Exception as ex:
    raise ex
    

File: Java/order/CreateOrderCheckout.java
package examples.order;

import me.pagar.api.PagarmeCoreApiClient;
import me.pagar.api.controllers.OrdersController;
import me.pagar.api.http.client.*;
import me.pagar.api.models.*;
import org.joda.time.DateTime;

import java.util.*;

public class CreateOrderCheckout {

    public static void main(String[] args) {

        // Configuration parameters and credentials
        String basicAuthUserName = "basicAuthUserName"; // The username to use with basic authentication
        String basicAuthPassword = "basicAuthPassword"; // The password to use with basic authentication

        PagarmeCoreApiClient client = new PagarmeCoreApiClient(basicAuthUserName, basicAuthPassword);

        OrdersController orders_controller = new OrdersController();

        CreateCustomerRequest create_customer_request = new CreateCustomerRequest();
        create_customer_request.setName("Tony Stark");
        create_customer_request.setEmail("tonystark@avengers.com");

        CreateCheckoutPaymentRequest create_checkout_payment_request = new CreateCheckoutPaymentRequest();
        create_checkout_payment_request.setCustomerEditable(false);
        create_checkout_payment_request.setSkipCheckoutSuccessPage(true);
        List<String> acceptedPaymentMethods = new ArrayList<String>();
        acceptedPaymentMethods.add("credit_card");
        acceptedPaymentMethods.add("boleto");
        acceptedPaymentMethods.add("bank_transfer");
        acceptedPaymentMethods.add("debit_card");
        create_checkout_payment_request.setAcceptedPaymentMethods(acceptedPaymentMethods);

        ArrayList<ArrayList<String>> multiPayments = new ArrayList<ArrayList<String>>();
        ArrayList<String> paymentOne = new ArrayList<String>();
        paymentOne.add("credit_card");
        paymentOne.add("credit_card");
        multiPayments.add(paymentOne);
        ArrayList<String> paymentTwo = new ArrayList<String>();
        paymentTwo.add("credit_card");
        paymentTwo.add("boleto");
        multiPayments.add(paymentTwo);
        create_checkout_payment_request.setAcceptedMultiPaymentMethods(Collections.singletonList(multiPayments));

        create_checkout_payment_request.setSuccessUrl("https://www.pagar.me'");

        create_checkout_payment_request.setBankTransfer(new CreateCheckoutBankTransferRequest());
        List<String> banksTransfer = new ArrayList<String>();
        banksTransfer.add("237");
        banksTransfer.add("001");
        banksTransfer.add("341");
        create_checkout_payment_request.getBankTransfer().setBank(banksTransfer);

        create_checkout_payment_request.setBoleto(new CreateCheckoutBoletoPaymentRequest());
        create_checkout_payment_request.getBoleto().setBank("033");
        create_checkout_payment_request.getBoleto().setInstructions("Pagar ate o vencimento");
        create_checkout_payment_request.getBoleto().setDueAt(DateTime.parse("2021-07-25T00:00:00Z"));

        create_checkout_payment_request.setCreditCard(new CreateCheckoutCreditCardPaymentRequest());
        create_checkout_payment_request.getCreditCard().setStatementDescriptor("Descriptor example");

        CreateCheckoutCardInstallmentOptionRequest create_checkout_card_installment_option_request_one = new CreateCheckoutCardInstallmentOptionRequest();
        create_checkout_card_installment_option_request_one.setNumber(1);
        create_checkout_card_installment_option_request_one.setTotal(2000);
        CreateCheckoutCardInstallmentOptionRequest create_checkout_card_installment_option_request_two = new CreateCheckoutCardInstallmentOptionRequest();
        create_checkout_card_installment_option_request_two.setNumber(1);
        create_checkout_card_installment_option_request_two.setTotal(2500);
        List<CreateCheckoutCardInstallmentOptionRequest> listInstallment = new ArrayList<CreateCheckoutCardInstallmentOptionRequest>();
        listInstallment.add(create_checkout_card_installment_option_request_one);
        listInstallment.add(create_checkout_card_installment_option_request_two);
        create_checkout_payment_request.getCreditCard().setInstallments(listInstallment);

        create_checkout_payment_request.setDebitCard(new CreateCheckoutDebitCardPaymentRequest());
        create_checkout_payment_request.getDebitCard().setAuthentication(new CreatePaymentAuthenticationRequest());
        create_checkout_payment_request.getDebitCard().getAuthentication().setType("threed_secure");
        create_checkout_payment_request.getDebitCard().getAuthentication()
                .setThreedSecure(new CreateThreeDSecureRequest());
        create_checkout_payment_request.getDebitCard().getAuthentication().getThreedSecure().setMpi("acquirer");
        create_checkout_payment_request.getDebitCard().getAuthentication().getThreedSecure()
                .setSuccessUrl("https://www.pagar.me");

        CreateOrderRequest request = new CreateOrderRequest();
        request.setCode("test-SDK-Java");

        CreateOrderItemRequest create_order_item_request = new CreateOrderItemRequest();
        create_order_item_request.setDescription("Tesseract Bracelet");
        create_order_item_request.setQuantity(1);
        create_order_item_request.setAmount(29090);
        List<CreateOrderItemRequest> listOrder = new ArrayList<CreateOrderItemRequest>();
        listOrder.add(create_order_item_request);
        request.setItems(listOrder);

        CreatePaymentRequest create_payment_request = new CreatePaymentRequest();
        create_payment_request.setAmount(2000);
        create_payment_request.setPaymentMethod("checkout");
        create_payment_request.setCheckout(create_checkout_payment_request);
        List<CreatePaymentRequest> listPaymentItems = new ArrayList<CreatePaymentRequest>();
        listPaymentItems.add(create_payment_request);
        request.setPayments(listPaymentItems);

        request.setCustomer(create_customer_request);

        orders_controller.createOrderAsync(request, null, new APICallBack<GetOrderResponse>() {
            @Override
            public void onSuccess(HttpContext context, GetOrderResponse response) {

                System.out.println("Order checkout create !");
                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println("Order result status: " + response.getStatus());
                System.out.println("Order id: " + response.getId());

            }

            @Override
            public void onFailure(HttpContext context, Throwable error) {

                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println(error.getMessage());
                error.printStackTrace();

            }
        });
    }
}